source/admin_manual/list_of_events.rst

```diff
@@ -1113,6 +1113,29 @@ IMAP Command
    .. Note:: This event is currently not sent for pre-login IMAP commands.
 
 
+.. dovecot_core:event:: imap_id_received
+   :added: v2.4.0;v3.0.0
+   :inherit: imap_client
```

Comment:
This is the event fields that I get from the pre-login event:
- local_ip
- local_port
- remote_ip
- remote_port
- service
- duration
- custom ID params

From the post-login event:
- user
- local_ip
- local_port
- remote_ip
- remote_port
- session
- duration
- cmd_name
- reason_code
- custom ID params

The diff is `service` here, and `user`, `session`, `cmd_name`, as well as `reason_code` there.

If the `imap_id_received` event should inherit from IMAP Command, the `cmd_tag`, `cmd_input_name`, `cmd_args` and `cmd_human_args` are missing from the event. It seems to be more in line with the `imap_client` group + the `cmd_name` field. I would love to adhere to your comment, but I feel like the inherit-change would mis-document the event fields.

Reply:
Looks like cmd_name comes from the global command event. I had forgotten those things also exist and modify events outside normal inheritance paths. And thinking further, maybe it's not so useful to inherit directly from command event. Mainly because there are no other events inheriting from it either.

But it is kind of problematic documenting cmd_name here, because it's not directly related to the event or its direct parents. It's only there because it's in a global event. If it was documented here, it would probably have to be documented for lots of other events as well. Also Marco is adding another cmd_tag to this global event. So if cmd_name is documented here, cmd_tag would have to be added here as well. All in all, I think it's just going to cause the documentation to become inconsistent and difficult to update properly. So I think better just remove it here. Perhaps some day we can figure out how to better document all this inheritance jungle.